apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -158,6 +158,20 @@ pub fn reset(config: config::Ledger) -> Result<()> {
     Ok(())
 }
 
+pub fn rollback(config: config::Ledger) -> Result<()> {
+    // Rollback Tendermint state
+    let tendermint_block_height =
```

Comment:
We could backup the Namada state here before trying the rollback to have a copy in case the procedure should fail and leave the db in an inconsistent state. From what I saw, the tendermint rollback command doesn't do that so we could do the same for the tendermint state